src/OrasProject.Oras/Registry/Remote/Repository.cs

```diff
@@ -369,6 +371,71 @@ internal Reference ParseReferenceFromContentReference(string reference)
     public async Task MountAsync(Descriptor descriptor, string fromRepository, Func<CancellationToken, Task<Stream>>? getContent = null, CancellationToken cancellationToken = default) 
         => await ((IMounter)Blobs).MountAsync(descriptor, fromRepository, getContent, cancellationToken).ConfigureAwait(false);
 
+    /// <summary>
+    /// PingReferrers returns true if the Referrers API is available for the repository,
+    /// otherwise returns false
+    /// </summary>
+    /// <param name="cancellationToken"></param>
+    /// <returns></returns>
+    /// <exception cref="ResponseException"></exception>
+    /// <exception cref="Exception"></exception>
+    internal bool PingReferrers(CancellationToken cancellationToken = default)
+    {
+        if (ReferrersState == Referrers.ReferrersState.Supported)
+        {
+            return true;
+        }
+
+        if (ReferrersState == Referrers.ReferrersState.NotSupported)
+        {
+            return false;
+        }
```

Comment:
nit: How about using a switch clause?
```suggestion
        switch (ReferrersState)
        {
            case Referrers.ReferrersState.Supported:
                return true;
            case Referrers.ReferrersState.NotSupported:
                return false;
        }
```